django/db/models/fields/files.py

```diff
@@ -312,6 +313,8 @@ def get_prep_value(self, value):
 
     def pre_save(self, model_instance, add):
         file = super().pre_save(model_instance, add)
+        if isinstance(file.name, EmptyName):
```

Comment:
Is there a reason why this can't simply be 

```suggestion
        if file.name is None:
```

What's the purpose of the `EmptyName` sentinel object?

Reply:
As mentioned,
> - Simply adding `if file.name is None:` would affect many other functioning codes.
>    - File fields with `null=True` option will fall into this statement as well

It will break 12 tests. Those tests consists creating an model instance that contains `FileField` with `null=True` option.

For example, in [test_render_options_fk_as_pk](https://github.com/django/django/blob/8c257cecffc314179a5375ab8a16ca0ff3b6fb16/tests/admin_widgets/test_autocomplete_widget.py#L145) test, one of which 12 breaking tests, `rubber_soul = Album.objects.create(name="Rubber Soul", band=beatles)` will not work with `if file.name is None:` condition because [`Album`](https://github.com/django/django/blob/8c257cecffc314179a5375ab8a16ca0ff3b6fb16/tests/admin_widgets/models.py#L55-L63) has `backside_art = MyFileField(upload_to="albums_back", null=True)`.

For that reason, As mentioned,
>Besides `None`, it is Needed something to represent a `ContentFile.name` is empty.

I introduced `EmptyName` sentinel object

Other options are welcomed.